bittide-rust-ffi/src/lib.rs

```diff
@@ -0,0 +1,530 @@
+// SPDX-FileCopyrightText: 2023 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![feature(slice_ptr_get)]
+use hs_bindgen::{traits::ReprRust, traits::ReprC, *};
+use std::mem;
+use std::fmt;
+
+#[macro_use]
+extern crate static_assertions;
+
+/// Rust sibling of `Bittide.ClockControl.SpeedChange`.
+#[repr(u32)]
+pub enum SpeedChange
+  { /// Increases clock speed.
+    SpeedUp  = 0
+  , /// Decreases clock speed.
+    SlowDown = 1
+  , /// Keeps the clock as it is.
+    NoChange = 2
+  }
+
+// We use `static_assert` to verify that the memory layout appears to
+// be as expected for ensuring a consistent encoding at the Haskell
+// side.
+const_assert!(
+  4 ==
+    mem::size_of::<SpeedChange>()
+);
```

Comment:
The Rust version we use you can do `const _: () = assert!(4 == mem::size_of::<SpeedChange>());` and get rid of the `static_assert` dependency.